src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseSign1Message.cs

```diff
@@ -5,372 +5,265 @@
-            }
-            else
+            if (IsDetached)
             {
-                Debug.Assert(keyType == KeyType.RSA);
-                bytesWritten = CoseHelpers.SignHashWithRSA((RSA)key, hasher, hashAlgorithm, buffer);
+                throw new CryptographicException(SR.Sign1VerifyContentWasDetached);
             }
 
-            writer.WriteByteString(buffer.Slice(0, bytesWritten));
+            return VerifyCore(key, _content, null, associatedData, CoseHelpers.GetKeyType(key));
         }
 
-        [UnsupportedOSPlatform("browser")]
-        public bool Verify(AsymmetricAlgorithm key)
+        public bool VerifyEmbedded(AsymmetricAlgorithm key, ReadOnlySpan<byte> associatedData)
         {
             if (key is null)
             {
                 throw new ArgumentNullException(nameof(key));
             }
 
-            if (_content == null)
+            if (IsDetached)
             {
                 throw new CryptographicException(SR.Sign1VerifyContentWasDetached);
             }
 
-            return VerifyCore(key, _content, null, GetKeyType(key));
+            return VerifyCore(key, _content, null, associatedData, CoseHelpers.GetKeyType(key));
         }
 
-        [UnsupportedOSPlatform("browser")]
-        public bool Verify(AsymmetricAlgorithm key, byte[] content)
+        public bool VerifyDetached(AsymmetricAlgorithm key, byte[] detachedContent, byte[]? associatedData = null)
         {
             if (key is null)
             {
                 throw new ArgumentNullException(nameof(key));
             }
-            if (content is null)
+            if (detachedContent is null)
             {
-                throw new ArgumentNullException(nameof(content));
+                throw new ArgumentNullException(nameof(detachedContent));
             }
 
-            if (_content != null)
+            if (!IsDetached)
             {
                 throw new CryptographicException(SR.Sign1VerifyContentWasEmbedded);
             }
 
-            return VerifyCore(key, content, null, GetKeyType(key));
+            return VerifyCore(key, detachedContent, null, associatedData, CoseHelpers.GetKeyType(key));
         }
 
-        [UnsupportedOSPlatform("browser")]
-        public bool Verify(AsymmetricAlgorithm key, ReadOnlySpan<byte> content)
+        public bool VerifyDetached(AsymmetricAlgorithm key, ReadOnlySpan<byte> detachedContent, ReadOnlySpan<byte> associatedData = default)
         {
             if (key is null)
             {
                 throw new ArgumentNullException(nameof(key));
             }
 
-            if (_content != null)
+            if (!IsDetached)
             {
                 throw new CryptographicException(SR.Sign1VerifyContentWasEmbedded);
             }
 
-            return VerifyCore(key, content, null, GetKeyType(key));
+            return VerifyCore(key, detachedContent, null, associatedData, CoseHelpers.GetKeyType(key));
```

Comment:
For the Verify methods, calling Embedded vs Detached backwards is probably better represented as an InvalidOperationException (since this method can't be called because of the state of the object (message.Content.HasValue is wrong for this method).